R/compute_scale.R

```diff
@@ -0,0 +1,115 @@
+#' Compute scale parameters
+#'
+#' Computes the average of a set of source values and transforms the result
+#' from the source range to the target range. For example, for calculating the
+#' average of a set of questionnaire response scores and re-coding the average
+#' response to obtain a subscale score.
+#'
+#' @param source A vector of values to be scaled
+#'
+#'   A numeric vector is expected.
+#'
+#' @param source_range The permitted source range
+#'
+#'   A numeric vector containing two elements is expected, representing the
+#'   lower and upper bounds of the permitted source range. Alternatively, if no
+#'   argument is specified for `source_range` and `target_range`, no
+#'   transformation will be performed.
+#'
+#' @param target_range The target range
+#'
+#'   A numeric vector containing two elements is expected, representing the
+#'   lower and upper bounds of the target range. Alternatively, if no
+#'   argument is specified for `source_range` and `target_range`, no
+#'   transformation will be performed.
+#'
+#' @param flip_direction Flip direction of the scale?
+#'
+#'   The transformed values will be reversed within the target range, e.g.
+#'   within the range 0 to 100, 25 would be reversed to 75.
+#'
```

Comment:
I would add that this argument is ignored if `source_range` and `target_range` are not specified.

Reply:
Good point, updated.